openqabot/types/incidents.py

```diff
@@ -76,204 +76,215 @@ def _is_scheduled_job(
-                        if not (neg and pos):
-                            full_post["qem"]["withAggregate"] = False
-
-                    # some arch specific packages doesn't have aggregate tests
-                    if not self.singlearch.isdisjoint(set(inc.packages)):
-                        full_post["qem"]["withAggregate"] = False
-
-                    delta_prio = data.get("override_priority", 0)
-
-                    if delta_prio:
-                        delta_prio -= 50
-                    else:
-                        if flavor.endswith("Minimal"):
-                            delta_prio -= 5
-                        if not inc.staging:
-                            delta_prio += 10
-                        if inc.emu:
-                            delta_prio = -20
-                        # override default prio only for specific jobs
-                        if delta_prio:
-                            full_post["openqa"]["_PRIORITY"] = BASE_PRIO + delta_prio
-
-                    # add custom vars to job settings
-                    if "params_expand" in data and any(
-                        forbidden_key in data["params_expand"]
-                        for forbidden_key in ["DISTRI", "VERSION"]
-                    ):
-                        log.error(
-                            "flavor:%s ignored as DISTRI and VERSION not allowed in params_expand",
-                            flavor,
-                        )
-                        continue
-
-                    if "params_expand" in data:
-                        full_post["openqa"].update(data["params_expand"])
-
-                    full_post["openqa"][
-                        "__SMELT_INCIDENT_URL"
-                    ] = f"https://smelt.suse.de/incident/{inc.id}"
-                    full_post["openqa"][
-                        "__DASHBOARD_INCIDENT_URL"
-                    ] = f"{QEM_DASHBOARD}incident/{inc.id}"
-
-                    settings = full_post["openqa"].copy()
-
-                    # if set, we use this query to detect latest public cloud tools image which used for running
-                    # all public cloud related tests in openQA
-                    if "PUBLIC_CLOUD_TOOLS_IMAGE_QUERY" in settings:
-                        settings = apply_pc_tools_image(settings)
-                        if not settings.get("PUBLIC_CLOUD_TOOLS_IMAGE_BASE", False):
-                            continue
-
-                    # parse Public-Cloud pint query if present
-                    if "PUBLIC_CLOUD_PINT_QUERY" in settings:
-                        settings = apply_publiccloud_pint_image(settings)
-                        if not settings.get("PUBLIC_CLOUD_IMAGE_ID", False):
-                            continue
-
-                    full_post["openqa"] = settings
-                    full_post["qem"]["settings"] = settings
-                    ret.append(full_post)
+        ret = list(filter(None, ret))
```

Comment:
please no ... this is awful and really discouraged python syntax

`ret = [r for r in ret if r]`

Reply:
thank you. That's actually also what I prefer. Done